optimum/habana/transformers/trainer.py

```diff
@@ -986,7 +986,10 @@ def hpu_deepspeed_checkpointing(function, *checkpoint_args, use_reentrant: Optio
                             inputs["flash_attention_causal_mask"] = True
                 if self.model.config is not None:
                     if self.model.config.model_type in ["llama", "qwen2", "mistral", "starcoder2"]:
-                        inputs["lazy_mode"] = args.use_lazy_mode
+                        forward_method = getattr(self.model, "forward")
```

Comment:
@vidyasiv since we only change CausalLM models for optimization to include lazy_model, should we specifically check for architecture include CausalLM ?

Reply:
cc: @yafshar

Reply:
@libinta  Yeah this fix is better, wondering if we should even remove the model_type if intention was to use lazy_mode for all models which support lazy_mode. For example, gemma2.